packages/frontend/app/pods/components/pod/application/off-canvas-container/sidebar/contacts/contact/component.ts

```diff
@@ -86,4 +97,15 @@ export default class SidebarContact extends Component<IArgs> {
 
     this.router.transitionTo('chat.privately-with', this.args.contact.id);
   }
+
+  @action onPinClick() {
```

Comment:
the action of pinning may not always be a click, it may be a spacebar or enter key press. 

maybe the action could simply be the verb `pin`?